### PR TITLE
Create buttons

### LINKED ISSUE #2: Clickability
Each cell of the grid should be clickable. When clicked, a box should appear containing a text input field.

### PR DESCRIPTION
**Pull request by kerznal**
Closes #2 I think.

### CODE REVIEW

File: ui/src/App.tsx [hunk 1]
```diff
@@ -22,6 +28,44 @@ const TABS: TabsType[] = ['today', 'yesterday', 'archived'];
...
+  function CreateButton(i: number){
+    const [open, setOpen] = useState(false);
+
+    const handleClickOpen = () => {
+      setOpen(true);
+    };
+
+    const handleClose = () => {
+      setOpen(false);
+    };
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
    const handleOpen = useCallback(() => setOpen(true), []);
    
    const handleClose = useCallback(() => setOpen(false), []);
```

See here:

https://react.dev/reference/react/useCallback

File: ui/src/App.tsx [hunk 2]
```diff
@@ -22,6 +28,44 @@ const TABS: TabsType[] = ['today', 'yesterday', 'archived'];
...
+          </DialogActions>
+        </Dialog>
+      </>
+    );
+  }
+
+  const grid_boxes = [];
+  for (let i = 0; i < 9; i++){
+    grid_boxes.push(CreateButton(i));
+  }
```
Comment (mattrafuse) [inline ui/src/App.tsx]: Javascript uses `camelCase`, not `snake_case`.

File: ui/src/App.tsx [hunk 3]
```diff
@@ -85,15 +130,7 @@ function App() {
...
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
+          {grid_boxes}
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
          {[...Array(9).keys()].map(entry => <SquareButton key={entry} />)}
```

File: ui/src/App.tsx [hunk 4]
```diff
@@ -22,6 +28,44 @@ const TABS: TabsType[] = ['today', 'yesterday', 'archived'];
 function App() {
   const [tab, setTab] = useState<TabsType>('today');
 
+  function CreateButton(i: number){
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
  const GridButton = () => {
```

Traditional function syntax has largely been replaced by lambdas: https://www.vinta.com.br/blog/javascript-lambda-and-arrow-functions

Also, generally components are moved into their own file, i.e. `GridButton.tsx` and then imported.

**Review by mattrafuse — CHANGES_REQUESTED** [review]

File: ui/src/App.tsx (hunk 1 above)
Comment (kerznal) [reply ui/src/App.tsx]: Makes sense. Applied this change in the new GridButton.tsx file.

File: ui/src/App.tsx (hunk 3 above)
Comment (kerznal) [reply ui/src/App.tsx]: Applied this change but using GridButton instead of SquareButton

**Review by rjarsenault1101 — APPROVED** [review]
yeah sure, I know what's going on here

File: ui/src/App.tsx [hunk 5]
```diff
@@ -2,11 +2,17 @@ import { useState } from 'react';
...
   Container,
+  // Dialog,
+  // DialogActions,
+  // DialogContent,
+  // DialogTitle,
   IconButton,
   Stack,
   Tab,
   Tabs,
+  // TextField,
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
  Container,
  IconButton,
  Stack,
  Tab,
  Tabs,
```

Always remove unused imports

File: ui/src/App.tsx [hunk 6]
```diff
@@ -15,13 +21,16 @@ import {
 import Refresh from '@mui/icons-material/Refresh';
 import { brown } from '@mui/material/colors';
 
+import gridButton from './components/GridButton';
```
Comment (mattrafuse) [inline ui/src/App.tsx]:
```suggestion
import GridButton from './components/GridButton';
```

Components are always Capitalized

File: ui/src/App.tsx [hunk 7]
```diff
@@ -85,15 +95,7 @@ function App() {
...
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
-          <Box></Box>
+          {[...Array(9).keys()].map(entry => gridButton(entry))}
```
Comment (mattrafuse) [inline ui/src/App.tsx]: This isn't how you call react components. Instead, you treat them like HTML tags:

`<GridButton key={entry} />`

File: ui/src/components/GridButton.tsx [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+const GridButton = (i: number) => {
+    const [open, setOpen] = useState(false);
+
+    const handleOpen = useCallback(() => setOpen(true), []);
+    
+    const handleClose = useCallback(() => setOpen(false), []);
+
+    return(
+      <>
+        <Button onClick={handleOpen} key={i}></Button>
```
Comment (mattrafuse) [inline ui/src/components/GridButton.tsx]: keys are always set in the context they are used in, i.e. When calling the component in the map function. You never pass them in.

https://react.dev/learn/rendering-lists

**Review by mattrafuse — CHANGES_REQUESTED** [review]

**Review by mattrafuse — APPROVED** [review]

**Pull request merged**